src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/utils/user_config_validatator.py

```diff
@@ -74,6 +101,60 @@ def block_DHCP_config(self):
         if "isStatic" in self.__config and not self.__config["isStatic"]:
             raise InvalidInputError("isStatic should be true. Only Static IP configuration is supported currently")
 
+    def check_if_old_nw_config(self):
+        if not self.__config["applianceControlPlaneIpAddress"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolStart"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["gatewayIPAddress"].strip():
+            return False
+        return True
+
+    def check_if_new_nw_config(self):
+        if self.__config["applianceControlPlaneIpAddress"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolStart"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["gatewayIPAddress"].strip():
+            return False
+        return True
+
+    def get_config_version(self):
+        if(self.check_if_old_nw_config()):
+            return Constant.OLD_CONFIG_VERSION
+        if(self.check_if_new_nw_config()):
+            return Constant.NEW_CONFIG_VERSION
+        raise InvalidInputError("Invalid Config provided")
+
+    def validate_segment_details_config(self, customer_resource):
+        customer_res: CustomerResource = customer_resource
+        res = self._segment_helper.get_segment_list(customer_res.subscription_id, 
+                                                    customer_res.resource_group, customer_res.private_cloud)
+        
+        segment_in_config = self.__config["staticIpNetworkDetails"]["networkForApplianceVM"]
```

Comment:
>segment_in_config [](http://example.com/codeflow?start=8&length=17)

Great to see additional config validation btw :)

nit - Would be good to mention what we are validating (in comments may be for now. If this list expands, we can modularize further then)

Reply:
sure